### DISCUSSION TITLE
How to set or determine sample rate of DMM?

### DISCUSSION THREAD

**Discussion by Skillmon** [ni/nimi-python#1662] [Q&A] [answered]
I was wondering, how could one set the sample of the supported DMMs. As far as I can tell, reducing the `nidmm.Session.aperture_time` doesn't have any effect. The used DMM-4082 should support >1 MS/s, but I get <1000 measurements per Minute and reducing the aperture time doesn't make any difference (I also set the averaging to 1).

**Comment by marcoskirsch** [comment, marked as answer]
Hello, for general questions about NI product usage, please use [NI's discussion forums](https://forums.ni.com/?profile.language=en).
In order to use your instrument as a digitizer, set [`operation_mode`](
https://nimi-python.readthedocs.io/en/master/nidmm/class.html#nidmm.Session.operation_mode) to [`WAVEFORM`](https://nimi-python.readthedocs.io/en/master/nidmm/enums.html#operationmode). Only then does ['waveform-rate'](https://nimi-python.readthedocs.io/en/master/nidmm/class.html#waveform-rate) apply.
